packages/cart/src/models/line-item.ts

```diff
@@ -0,0 +1,168 @@
+import { DAL } from "@medusajs/types"
+import { generateEntityId } from "@medusajs/utils"
+import {
+  BeforeCreate,
+  Cascade,
+  Collection,
+  Entity,
+  Index,
+  ManyToOne,
+  OnInit,
+  OneToMany,
+  OptionalProps,
+  PrimaryKey,
+  Property,
+} from "@mikro-orm/core"
+import Cart from "./cart"
+import LineItemAdjustmentLine from "./line-item-adjustment-line"
+import LineItemTaxLine from "./line-item-tax-line"
+
+type OptionalLineItemProps =
+  | "is_discoutable"
+  | "is_tax_inclusive"
+  | "compare_at_unit_price"
+  | "requires_shipping"
+  | DAL.EntityDateColumns
+
+@Entity({ tableName: "cart_line_item" })
+export default class LineItem {
+  [OptionalProps]?: OptionalLineItemProps
+
+  @PrimaryKey({ columnType: "text" })
+  id: string
+
+  @ManyToOne(() => Cart, {
+    onDelete: "cascade",
+    index: "IDX_line_item_cart_id",
+    fieldName: "cart_id",
+  })
+  cart!: Cart
+
+  @Property({ columnType: "text" })
+  title: string
+
+  @Property({ columnType: "text", nullable: true })
+  subtitle: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  thumbnail: string | null
+
+  @Property({ columnType: "text" })
+  quantity: number
+
+  @Index({
+    name: "IDX_line_item_variant_id",
+    properties: ["variant_id"],
+  })
+  @Property({ columnType: "text", nullable: true })
+  variant_id: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_id: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_title: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_description: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_subtitle: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_type: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_collection: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  product_handle: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  variant_sku: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  variant_barcode: string | null
+
+  @Property({ columnType: "text", nullable: true })
+  variant_title: string | null
+
+  @Property({ columnType: "jsonb", nullable: true })
+  variant_option_values: Record<string, unknown> | null
```

Comment:
suggestion: I'm wondering if we could have another table to relate the property (id) and the value. That way we could have a more generic cart system not mapping product properties.

`item_property`
`line_item`
`line_item_item_property`

Reply:
Having the variant and product properties specifically was meant to solve for two things:
- Model properties that are easy to work with
- Cart line items similar to how many other providers define them. E.g. Shopify and commercetools.

What other properties do you think could be relevant to have – potentially via the line item property approach?

Reply:
Wil merge this now, so I can move forward.

But I will not commit any migrations yet nor will I work with items yet, so we have time next week to revisit :)